### DISCUSSION TITLE
Statistical weights of umbrella sampling trajectories (to analyze trajectories)

### DISCUSSION THREAD

**Discussion by srdjan125** [plumed/plumed2#884] [Q&A] [answered]
Hello, I am currently doing PLUMED tutorial for umbrella sampling (Lugano tutorial) and I have successfully calculated statistical weights of each US snapshot by applying wham.py program according to tutorials.

It was said that these weights can be used to calculate any property from US trajectory, not just free energy as a function of CV. Regarding free energy as a function of CV, this is clear and I can even do it by gmx wham command., but gmx wham cannot give me weights of each snapshot (to do other analysis).

I am trying to understand how should I apply these weights to compute some other quantities from US trajectory. Let's say that I want to compute only Coloumb electrostatic energies between molecule A and B, as a function of COM (center of mass distance between molecules A and B). Lets say at COM=5 nm Coloumb electrostatic energy can have 100 different values (due to the fact that many US snapshots can have COM=5nm and give different values due to different relative orientations between molecules).
So basically, if I want to determine Coloumb energy at 5nm , I need to find weighted average of those 100 values: weight1Couloumb1+weight2Coloumb2+ ..... + weight_n*Couloumb_n where weights are statistical weights of each snapshot obtained by wham.py

Is this correct way to how to compute random property from US trajectory? Or I need to apply some more soffisticated procedure?
I would appreciate if someone could help me with this issue. If this Discussion board is not appropriate please let me know about other suitable forum to ask this question.

Thank you in advance.

**Comment by GiovanniBussi** [comment, marked as answer]
If I understand correctly your procedure, it makes sense except that you should normalize the weights:

(weight1Couloumb1+weight2Coloumb2+ ..... + weight_n*Couloumb_n) /( weight1+weight2+ ..... + weight_n)

Indeed, the sum of the weights at each distance will be different.

**Reply by srdjan125** [reply]
Hello, thank you for your answer. Yes, that was another part of my dilemma, whether I should normalize weights at each COM distance separately (otherwise if I use "global" weights of whole US trajectory it might happen that there is more US snapshots having COM=4nm than COM=9nm , so it would artificially shift values of energy at some COM distances). 

I suppose the same re-weighting procedure should be applied even if I want to calculate frequencies of contacts between residues, or number of H-bonds? (and not just if I want to compute LJColoumb/torsional energies)

**Reply by GiovanniBussi** [reply]
yes
